esrally/mechanic/provisioner.py

```diff
@@ -184,7 +186,17 @@ def _provisioner_variables(self):
         plugin_variables = {}
         mandatory_plugins = []
         for installer in self.plugin_installers:
-            mandatory_plugins.append(installer.plugin_name)
+            # For Elasticsearch < 6.3 more specific plugin names are required for mandatory plugin check
+            # Details in: https://github.com/elastic/elasticsearch/pull/28710
+            try:
```

Comment:
I understand that it is a necessity at the moment but I think we should try to be version-agnostic in provisioner code. When earlier 6.x versions are not in use anymore I think we should remove this again to stay version-agnostic.

Reply:
I agree with you. I was hesitant to add this conditional here and wasn't 100% sure in which place to define the constraints themselves. Should I add a `TODO:` to remove this when <6.3 is EOL?

Reply:
yes, please add a TODO.